### PR TITLE
check_format: automatically infer API headers

### PR DESCRIPTION
**Pull request by htuch**
* Automatically infer Envoy API header files from source in envoy_build_fixer.py.
    
* Also remove unusued loads in envoy_build_fixer.py via Buildozer, we can enable other cleanups potentially in the future.

Signed-off-by: Harvey Tuch <htuch@google.com>

### CODE REVIEW

**Comment by lizan** [comment]
LGTM modulo the test failures.

File: source/extensions/filters/common/ext_authz/BUILD [hunk 1]
```diff
@@ -42,12 +42,6 @@ envoy_cc_library(
         "//source/common/network:utility_lib",
         "//source/common/protobuf",
         "//source/common/tracing:http_tracer_lib",
-
-        # TODO(gsagula): Descriptor pool requires this dependence in runtime only. It should NOT be
-        # removed unless v2alpha is fully deprecated, otherwise linker will drop the object file
-        # associated with v2alpha/external_auth.pb.cc causing FindMessageTypeByName() to always return
-        # nullptr when looking up for envoy.service.auth.v2alpha.Authorization.Check.
-        "@envoy_api//envoy/service/auth/v2alpha:pkg_cc_proto",
```
[the remark this answers is not in the document]
Comment (htuch) [reply source/extensions/filters/common/ext_authz/BUILD]: This PR is largely agnostic to whether something is deprecated. If it's not referenced anymore, it gets removed from the BUILD :)

Re: splitting the PR, yes, this is a reasonable ask. I was hoping for a sneaky combined review, but not this time around..

[the remark this answers is not in the document]
Comment (htuch) [reply source/extensions/filters/common/ext_authz/BUILD]: Split out the Python 3 and Buildozer bits in https://github.com/envoyproxy/envoy/pull/8663.

**Comment by jmarantz** [comment]
does this PR need to wait for #8663 until it gets small enough to review?

**Comment by htuch** [comment]
@jmarantz yep

**Comment by htuch** [comment]
Merged with  #8663, squashed and force pushed (since I assume there was no substantial review of this PR until now).

**Comment by htuch** [comment]
Friendly ping, PTAL.

**Comment by htuch** [comment]
@lizan I think this should be fixed (based on local execution), PTAL when you get a chance, thanks.

**Pull request merged**